src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -761,10 +769,16 @@ private void GetTypeNamespaceNamesOrThrow(Dictionary<string, ArrayBuilder<TypeDe
             {
                 string @namespace = MetadataReader.GetString(kvp.Key);
 
-                ArrayBuilder<TypeDefinitionHandle> builder;
+                ArrayBuilder<TypeDefinitionHandle>? builder;
 
                 if (namespaces.TryGetValue(@namespace, out builder))
                 {
+                    if (builder is null)
+                    {
+                        builder = new ArrayBuilder<TypeDefinitionHandle>(kvp.Value.Count);
+                        namespaces[@namespace] = builder;
+                    }
```

Comment:
Wanted to discuss this behavior change as I'm unsure what the best approach is here. It's easy to see that ` GetForwardedTypeNamespaceNamesOrThrow` inserts `null` values into the map here. There is nothing obvious I can see where these `null` values are ignored / removed hence the compiler warning on `builder` being potentiall `null` here has merit. I went back and forth on `Debug.Assert` or handle `null`. Definitely want to discuss this part.

Reply:
But GetForwardedTypeNamespaceNamesOrThrow is called after this method (GetTypeNamespaceNamesOrThrow), so there shouldn't be `null`s in the map at this point, right?